samples/integration/cmake-vcpkg/README.md

```diff
@@ -0,0 +1,532 @@
+- Now lets try building the project with:
+
+```bash
+cmake --build ./build
+```
+
+- The output of the build should contain a line stating where the executable that was built was placed. By default it should be in your `build` directory under a new `Debug` directory with the name `azure_sample.exe` if your on Windows or `azure_sample` if your on MacOS or Linux.
+- Lets try running our new executable.
+  - If your on Windows Powershell, enter:
+  
+  ```pwsh
+  .\build\Debug\azure_sample.exe
+  ```
+
+  - If your on MacOS or Linux, enter:
+  
+  ```bash
+  ./build/azure_sample
+  ```
+
+- You should get the following out put:
+
+```bash
+Hello World!
+```
+
+### Install packages with vcpkg
+
+- In this example we'll be using vcpkg in what's known as manifest mode.
+- First we'll start by moving back to the root directory of our project, and creating a new vcpkg application with the following command:
+
+```bash
+vcpkg new --application
+```
+
+- You should now have a `vcpkg.json` and `vcpkg-configuration.json` file in your project directory with the following contents:
+  - `vcpkg.json`
+  
+  ```json
+  {}
+  ```
+
+  - `vcpkg-configuration.json` *It's okay if the hash on the `baseline` property doesn't match exactly.*
+  
+  ```json
+  {
+    "default-registry": {
+      "kind": "git",
+      "baseline": "6af584dd59aa5bdba75dae6781ec74614e03e5b9",
+      "repository": "https://github.com/microsoft/vcpkg"
+    },
+    "registries": [
+      {
+        "kind": "artifact",
+        "location": "https://github.com/microsoft/vcpkg-ce-catalog/archive/refs/heads/main.zip",
+        "name": "microsoft"
+      }
+    ]
+  }
+  ```
+
+- Now we can add the Azure Key Vault and Identity libraries from the Azure SDK for C++ to our project, by entering the following command:
+
+```bash
+vcpkg add port azure-identity-cpp azure-security-keyvault-secrets-cpp
+```
+
+- Your `vcpkg.json` file should now look like this:
+
+```json
+{
+  "dependencies": [
+    "azure-identity-cpp",
+    "azure-security-keyvault-secrets-cpp"
+  ]
+}
+```
+
+### Integrate CMake with vcpkg
+
+- To integrate CMake with vcpkg we need to add the `vcpkg.cmake` module to our CMake toolchain. You can do this either by setting `CMAKE_TOOLCHAIN_FILE` in your `CMakeLists.txt` file like this:
+
+```cmake
+cmake_minimum_required(VERSION 3.30.0)
+
+# Remember to replace the path below with the path where you cloned vcpkg
+set(CMAKE_TOOLCHAIN_FILE "/path/to/vcpkg-root/scripts/buildsystems/vcpkg.cmake")
+
+project(azure_sample VERSION 0.1.0 LANGUAGES C CXX)
+
+add_executable(azure_sample main.cpp)
+```
+
+- Or by specifying the toolchain file in the CMake configuration command like this:
+
+```bash
+cmake -DCMAKE_TOOLCHAIN_FILE=/path/to/vcpkg-root/scripts/buildsystems/vcpkg.cmake -B build
+```
+
+- I'd recommend specifying the toolchain file in the `CMakeLists.txt` for now as it will shorten the commands you have to enter.
```

Comment:
Do you want to discuss cmakepresets or cmakeuserpresets here? That is also a great place to stick the toolchain file definition.